### PR TITLE
feat: inbound RabbitMQ Connector

### PR DESCRIPTION
**Pull request by chillleader**
## Description

The authentication properties are basically copied from the outbound connector.
New properties:
- Queue name
- Consumer tag
- Subscription arguments
- Exclusive (yes/no)

Screenshot 2023-04-14 at 16 46 02
Screenshot 2023-04-14 at 16 46 19

## Error handling

- Reject and requeue - on unexpected errors (Zeebe became unavailable, any runtime errors other than `ConnectorInputException`)
- Reject and DON'T requeue:
  - On deserialization errors (caught as `ConnectorInputException`)
  - On FEEL expression evaluation error (caught as `ConnectorInputException`)
  - When activation condition wasn't met

## Related issues

related to https://github.com/camunda/team-connectors/issues/351

⚠️ depends on https://github.com/camunda/connector-sdk/pull/435

### CODE REVIEW

File: connectors/rabbitmq/element-templates/rabbitmq-start-event-connector.json [hunk 1]
```diff
@@ -0,0 +1,269 @@
+{
+  "$schema": "https://unpkg.com/@camunda/zeebe-element-templates-json-schema/resources/schema.json",
+  "name": "RabbitMQ Connector",
+  "id": "io.camunda.connectors.inbound.RabbitMQ.StartEvent.v1",
+  "version": 1,
+  "description": "Receive a message from RabbitMQ",
+  "documentationRef": "https://docs.camunda.io/docs/components/connectors/out-of-the-box-connectors/rabbitmq",
```
[the remark this answers is not in the document]
Comment (chillleader) [reply connectors/rabbitmq/element-templates/rabbitmq-start-event-connector.json:7]: I think so, but maybe I'll reorganize the doc a bit, in that case I will also fix it here.

File: bundle/mvn/camunda-saas-bundle/src/main/resources/application.properties [hunk 2]
```diff
@@ -17,5 +17,8 @@ camunda.connector.auth.issuer=https://weblogin.cloud.dev.ultrawombat.com/
 # Enforce local connection, even if cluster-id set (for Operate Auth)
 zeebe.client.connection-mode=ADDRESS
 
+# Enable operate client and secret provider initialization. Must be disabled (set to false) to avoid creating the default operate client bean.
+operate.client.enabled=false
+
```
Comment (sbuettner) [inline bundle/mvn/camunda-saas-bundle/src/main/resources/application.properties:22]: Duplicate?

File: connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/common/model/RabbitMqRouting.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+package io.camunda.connector.rabbitmq.common.model;
+
+import io.camunda.connector.api.annotation.Secret;
+import java.util.Objects;
+
+public class RabbitMqRouting {
+
+  @Secret protected String virtualHost;
+  @Secret protected String hostName;
+  @Secret protected String port;
```
Comment (sbuettner) [inline connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/common/model/RabbitMqRouting.java:16]: Just curious, do they need to use `@Secret`?

Comment (sbuettner) [reply connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/common/model/RabbitMqRouting.java:16]: Right, env vars are the missing piece of the puzzle.

File: connectors/rabbitmq/element-templates/rabbitmq-start-event-connector.json [hunk 4]
```diff
@@ -0,0 +1,269 @@
+{
+  "$schema": "https://unpkg.com/@camunda/zeebe-element-templates-json-schema/resources/schema.json",
+  "name": "RabbitMQ Connector",
```
[the remark this answers is not in the document]
Comment (chillleader) [reply connectors/rabbitmq/element-templates/rabbitmq-start-event-connector.json:3]: I think we don't necessarily have to: the outbound connector is only shown in the Modeler when the user picks a `Task/ServiceTask`. The inbound connector is only shown when the user picks `StartEvent`. Therefore there are no collisions between these names.

[the remark this answers is not in the document]
Comment (igpetrov) [reply connectors/rabbitmq/element-templates/rabbitmq-start-event-connector.json:3]: That makes sense

File: connectors/rabbitmq/src/test/java/io/camunda/connector/rabbitmq/integration/RabbitMqIntegrationTest.java
Comment (chillleader) [inline connectors/rabbitmq/src/test/java/io/camunda/connector/rabbitmq/integration/RabbitMqIntegrationTest.java:1]: Added an integration test with a real RabbitMQ container

File: connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/inbound/RabbitMqConsumer.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    }
+  }
+
+  private RabbitMqInboundResult prepareVariables(
+      String consumerTag, Envelope envelope, BasicProperties properties, byte[] body) {
+
+    try {
+      String bodyAsString = new String(body, StandardCharsets.UTF_8);
+      JsonElement bodyAsJsonElement =
+          GsonSupplier.gson()
```
Comment (sbuettner) [inline connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/inbound/RabbitMqConsumer.java:92]: Would be great to get rid of gson.

File: connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/inbound/RabbitMqConsumer.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        JsonPrimitive bodyAsPrimitive = (JsonPrimitive) bodyAsJsonElement;
+        if (bodyAsPrimitive.isBoolean()) {
+          bodyAsObject = bodyAsPrimitive.getAsBoolean();
+        } else if (bodyAsPrimitive.isNumber()) {
+          bodyAsObject = bodyAsPrimitive.getAsNumber();
+        } else {
+          bodyAsObject = bodyAsPrimitive.getAsString();
+        }
+      } else {
+        bodyAsObject = GsonSupplier.gson().fromJson(bodyAsJsonElement, Object.class);
```
Comment (sbuettner) [inline connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/inbound/RabbitMqConsumer.java:106]: see comment above

**Review by sbuettner — APPROVED** [review]
Looks good to me. Would be great to have a simple e2e test using testcontainers but this can be done in another PR as well.

**Review by sbuettner — APPROVED** [review]

File: connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/inbound/RabbitMqConsumer.java (hunk 5 above)
Comment (chillleader) [reply connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/inbound/RabbitMqConsumer.java:92]: Let's do it in a separate PR for all connectors

**Pull request merged**